sound/soc/sof/ipc4-topology.c

```diff
@@ -17,6 +17,11 @@
 #include "ipc4-topology.h"
 #include "ops.h"
 
+static bool force_cpc_to_zero;
+module_param_named(ipc4_force_cpc_to_zero, force_cpc_to_zero, bool, 0444);
+MODULE_PARM_DESC(ipc4_force_cpc_to_zero,
+		 "Ignore CPC value for modules and force it to be 0.");
```

Comment:
This is a bit hard for outsiders as you'd have to know that zero has special meaning to make any sense of this. Additionally CPC should be explains somehow. 

But, but, I don't really have very good alternatives. One solution:

```
MODULE_PARM_DESC(ipc4_ignore_fw_cpc,
    "Ignore the cycle budget (CPC) data for modules" 
```

Reply:
> This is a bit hard for outsiders as you'd have to know that zero has special meaning to make any sense of this. Additionally CPC should be explains somehow.

Are we going to document IPC4, clock scaling, clock sources and all of the stuff around CPC?

> 
> But, but, I don't really have very good alternatives. One solution:
> 
> ```
> MODULE_PARM_DESC(ipc4_ignore_fw_cpc,
>     "Ignore the cycle budget (CPC) data for modules" 
> ```
CPC == Cycle Budget?

Initially I had this: ipc4_ignore_cpc
My problem was and still is (after I thought about it) that the 'ignore' implies that we would do, use, rely on a 'cpc' value, but we are just reading it from manifest (at the moment, but we might have topology override and in multiple levels).
So we should have 'ipc4_ignore_queried_cpc_value', but ignoring does not tell that we are going to send 0 or cpc_max or 42.
I'm not sure if the added 'fw' helps in this.

Sure, let's roll a dice, I'm fine with whatever name ;)

Reply:
inline comments are free here, so we can be a little more verbose and say "... in other words forces DSP clock to max frequency"

Reply:
"force_cpc_to_zero" can only be understood if the reviewer *knows* that this zero value disables clock scaling.

If we want to avoid a description, 'ignore' is better IMHO. ipc4_ignore_fw_manifest_cpc ?

Reply:
it is going to ignore any CPC regardless of source and it might include in the future CPC from topology.
Should we go with a really generic `ipc4_ignore_cpc`.
I don't think we can expect any user to know what CPC is and what it does by the name of the parameter.